src/sdam/server_description.ts

```diff
@@ -82,6 +82,12 @@ export class ServerDescription {
     hello?: Document,
     options: ServerDescriptionOptions = {}
   ) {
+    if (address == null || address === '') {
+      throw new MongoInvalidArgumentError(
```

Comment:
should this be a `MongoRuntimeError` instead if it's not meant to be directly instantiated by users? I'm actually confused why the class is marked `public` but the comment above it says "internal type" and then this constructor is marked `@internal`... what are the rules?

Reply:
The class's **type** is public because it can be encountered through SDAM event emitters, but it is not meant to be constructed by anyone else so we mark the constructor internal (similar patterns are in our events, Db, and Collection classes). So you're correct this should be a runtime error, our driver would have to accidentally have missed some critical validation to encounter this.

```ts
var mdb = require('mongodb')
mdb.ServerDescription; // undefined
```